tests/regression/test_nullspace.py

```diff
@@ -119,6 +119,43 @@ def test_nullspace_mixed():
     assert sqrt(assemble((u - exact)*(u - exact)*dx)) < 5e-8
 
 
+def test_near_nullspace():
+    mesh = UnitSquareMesh(100, 100)
+    dim = 2
+    parameters["pyop2_options"]["block_sparsity"] = False
+    V = VectorFunctionSpace(mesh, "Lagrange", 1)
+    u = TrialFunction(V)
+    v = TestFunction(V)
+
+    mu = Constant(0.2)
+    lmbda = Constant(0.3)
+
+    def sigma(fn):
+        return 2.0 * mu * sym(grad(fn)) + lmbda * tr(sym(grad(fn))) * Identity(dim)
+
+    w_exact = Function(V)
+    w_exact.interpolate(Expression(("x[0]*x[1]", "x[0]*x[1]")))
+    # div(sigma(w_exact)) = (mu + lmbda, mu + lmbda)
+    f = Constant((mu + lmbda, mu + lmbda))
+    F = inner(sigma(u), grad(v))*dx + inner(f, v)*dx
+
+    bcs = [DirichletBC(V, w_exact, (1, 2, 3, 4))]
+
+    n0 = Constant((1, 0))
+    n1 = Constant((0, 1))
+    n2 = Expression(("x[1] - 0.5", "-(x[0]-0.5)"))
+    ns = [n0, n1, n2]
+    n_interp = [interpolate(n, V) for n in ns]
+    n_normalized = [interpolate(n*(1.0/sqrt(n.dat.inner(n.dat))), V) for n in n_interp]
+    nsp = VectorSpaceBasis(vecs=n_normalized)
+
+    w = Function(V)
+    solve(lhs(F) == rhs(F), w, bcs=bcs, solver_parameters={'ksp_monitor': True, 'ksp_rtol': 1e-8, 'ksp_atol': 1e-8, 'ksp_type': 'cg', 'pc_type': 'gamg'}, near_nullspace=nsp)
+    w = Function(V)
+    solve(lhs(F) == rhs(F), w, bcs=bcs, solver_parameters={'ksp_monitor': True, 'ksp_rtol': 1e-8, 'ksp_atol': 1e-8, 'ksp_type': 'cg', 'pc_type': 'gamg'})
+    assert sqrt(assemble(inner(w-w_exact, w-w_exact)*dx)) < 1e-7
```

Comment:
How do I determine if this worked?  I presume when you pass the near null space the convergence is much better?  Can you assert that, maybe by comparing the convergence against "known good" values.  You can see an example of how to do that at https://github.com/firedrakeproject/firedrake/blob/master/tests/regression/test_matrix_free.py#L59.  Where I test that two ostensibly equivalent solves produce the same convergence history.